### PR TITLE
accept yep 111 properties

### PR DESCRIPTION
**Pull request by untzag**
has been in use for a while

### CODE REVIEW

File: yeps/yep-111.md [hunk 1]
```diff
@@ -23,97 +23,110 @@ Fields are inspired by Bluesky, which has a system of [fields](http://nsls-ii.gi
...
 
-Here "field" is a single piece of information about a daemon, things like "current position", "integration time", and "averaging method".
-This YEP allows daemons to help connected clients by specifying relevant fields in a standardized way.
-While the primary purpose of fields is automatic user-interface generation, fields may also be useful when choosing what (meta)data to record from each daemon in the context of an experiment.
+Here "property" is a single piece of information about a daemon, things like "current position", "integration time", and "averaging method".
+This YEP allows daemons to help connected clients by specifying relevant properties in a standardized way.
+While the primary purpose of properties is automatic user-interface generation, fields may also be useful when choosing what (meta)data to record from each daemon in the context of an experiment.
 
 Fields exist on-top-of and do not modify or define the Avro-RPC protocol itself.
 Clients may ignore fields without penalty.
```
Comment (ksunden) [inline yeps/yep-111.md:48]:
```suggestion
Clients may ignore properties without penalty.
```

File: yeps/yep-111.md [hunk 2]
```diff
@@ -23,97 +23,110 @@ Fields are inspired by Bluesky, which has a system of [fields](http://nsls-ii.gi
...
+## The Properties Record
 
-Fields appear under the `fields` key of every protocol provided by yaq daemons.
+Properties appear under the `properties` key of every protocol provided by yaq daemons.
 This key maps to a `string: record` mapping where the string key is the field name and the record matches the following:
 
 ```
 {
   "type": "record",
   "name": "field",
```
Comment (ksunden) [inline yeps/yep-111.md:60]:
```suggestion
  "name": "property",
```

File: yeps/yep-111.md [hunk 3]
```diff
@@ -23,97 +23,110 @@ Fields are inspired by Bluesky, which has a system of [fields](http://nsls-ii.gi
...
   "fields": [
+    {"name": "type", "type": "string"}
     {"name": "getter", "type": "string"},
     {"name": "setter", "type": ["null", "string"], "default": "null"},
+    {"name": "units_getter", "type": ["null", "string"], "default": "null"},
+    {"name": "limits_getter", "type": ["null", "string"], "default": "null"},
```
Comment (ksunden) [inline yeps/yep-111.md:66]: missing options_getter, both here and in the prose

File: yeps/yep-111.md [hunk 4]
```diff
@@ -23,97 +23,110 @@ Fields are inspired by Bluesky, which has a system of [fields](http://nsls-ii.gi
...
     {"name": "dynamic", "type": "boolean", "default": true},
-    {"name": "kind", "type": "enum", "symbols": ["normal", "hinted"]},
-    {"name": "type", "type": "string"}
+    {"control_kind": "type": {"type": "enum", "name": "kind", "symbols": ["normal", "hinted"]},
+    {"record_kind": "type": {"type": "enum", "name": "kind", "symbols": ["data", "metadata"]},
```
Comment (ksunden) [inline yeps/yep-111.md:69]: missing `"omitted"`

File: yeps/yep-111.md [hunk 5]
```diff
@@ -23,97 +23,110 @@ Fields are inspired by Bluesky, which has a system of [fields](http://nsls-ii.gi
...
-## Fields and Traits
+## Properties and Traits
 
-Fields may be specified by traits.
-Individual daemons may overload trait-specified fields, but only in the following cases:
+Properties may be specified by traits.
+Individual daemons may overload trait-specified properties, but only in the following cases:
 
 - to change `dynamic` from `true` to `false`
 - to narrow the scope of a union type
```
Comment (ksunden) [inline yeps/yep-111.md:106]: provide limits getter? technically done via a trait for "position", but hmmmm

File: yeps/yep-111.md (hunk 1 above)
Comment (kameyer226) [reply yeps/yep-111.md:48]: What does it mean to ignore here?

Comment (untzag) [reply yeps/yep-111.md:48]: Thanks for the question @kameyer226 

I'm trying to point out that properties are fully optional. A client NEEDS to pay attention to the messages, as their description is needed to encode commands. Simple clients can choose to not use the additional metadata of properties.

**Review by ksunden — APPROVED** [review]

**Pull request merged**